sessionctx/variable/session.go

```diff
@@ -958,6 +958,8 @@ type SessionVars struct {
 		data [2]stmtctx.StatementContext
 	}
 
+	// EnableStmtOptimizeTrace indicates whether enable optimizer trace by 'trace plan statement'
+	EnableStmtOptimizeTrace bool
```

Comment:
As `ResetContextOfStmt` would reset `StmtCtx` before executing statement, we have to keep the optimize trace flag in `SessionVars` from `stmtCtx`.